v2_1/ws/rest/docs/4_4_data_queries.md

```diff
@@ -1,91 +1,104 @@
+context | One of the following: `datastructure`, `dataflow`, `provisionagreement` | Data can be reported against a data structure, a dataflow or a provision agreement. This parameter allows selecting the desired context for data retrieval.
+agencyID | A string compliant with the SDMX *common:NCNameIDType* | The agency maintaining the artefact for which data have been reported. 
+resourceID | A string compliant with the SDMX *common:IDType* | The id of the artefact for which data have been reported.
+version | A string compliant with the *VersionType* defined in the SDMX Open API specification | The version of the artefact for which data have been reported.
+key | A string compliant with the *KeyType* defined in the SDMX Open API specification. | The combination of dimension values identifying the slice of the cube for which data should be returned. Wildcarding is supported via the `*` operator. For example, if the following key identifies the bilateral exchange rates for the daily US dollar exchange rate against the euro, D.USD.EUR.SP00.A, then the following key can be used to retrieve the data for all currencies against the euro: D.*.EUR.SP00.A.
 
 The parameters mentioned above are specified using the following syntax:
 
-    protocol://ws-entry-point/resource/flowRef/key/providerRef
-    
-Furthermore, some keywords may be used:
-
-Keywords | Scope | Description
---- | --- | ---
-all | key | Returns all data belonging to the specified dataflow and provided by the specified provider.
-all | providerRef | Returns all data matching the supplied key and belonging to the specified dataflow that has been provided by any data provider. As `all` is a reserved keyword in the SDMX RESTful API, it is recommended not to use it as an identifier for providers.
+    protocol://ws-entry-point/resource/context/agencyID/resourceID/version/key
 
 The following rules apply:
 
-- If no key is specified, all data (or metadata) belonging to the dataflow (or metadataflow) identified by the flowRef should be supplied. It is therefore equivalent to using the keyword `all`.
-- If no providerRef is specified, the matching data (or metadata) provided by any data provider should be returned. It is therefore equivalent to using the keyword `all`.
+- All parameters support multiple values, using comma (`,`) as separator.
+- All parameters offer the option to retrieve all existing values, using the `*` operator.
```

Comment:
Except 'resource', 'context' and 'resourceID'. Maybe use similar formulation as suggested in availability review?

Reply:
`resource` is limited to `data` and is not in the list of path parameters mentioned in that section.

My understanding is that, for data queries, it is possible to have multiple values for context and resourceID as well, and this includes the possibility to use `*`. This allows making queries such as: "Give me all the data you have about Switzerland": https://ws-entry-point/data?c[REF_AREA]=CH, which is the short form of https://ws-entry-point/data/*/*/*/*/*?c[REF_AREA]=CH

Reply:
Wow! With this https://ws-entry-point/data you could then get all data of the whole dissemination database in one single shot! Maybe some data aggregators/resellers might love this but I'm not sure that we would be able to technically and securely support such a web service feature. Is this really needed by organisations or just nice to have? I could see use cases where you want to have data from several Dataflows or several DPs, but not really for 'all'. But I guess that this might only be generic and thus simple SDMX syntax and that implementers might choose to not implement specific features like 'all' for certain parameters.

If the 'context' is wildcarded (or has multiple selections) and the same data are defined for DSD, Dataflow and PA, would the data then have to be returned multiple times, each time in its context? If not, which context should be used by default or in preference?

Ok for the 'resource' but maybe it would be even clearer if you say "All path parameters support ..." and "All path parameters offer the option ..."?